tf/edgeml/trainer/bonsaiTrainer.py

```diff
@@ -389,18 +432,61 @@ def train(self, batchSize, totalEpochs, sess,
 
             # This helps in direct testing instead of extracting the model out
 
-            testAcc, testLoss, regTestLoss = sess.run(
-                [self.accuracy, self.loss, self.regLoss], feed_dict=_feed_dict)
+            testAcc, testLoss, regTestLoss, pred = sess.run(
+                [self.accuracy, self.loss, self.regLoss, self.prediction], feed_dict=_feed_dict)
+
             if ihtDone == 0:
-                maxTestAcc = -10000
-                maxTestAccEpoch = i
-            else:
-                if maxTestAcc <= testAcc:
+                if (self.bonsaiObj.isRegression == False):
+                    maxTestAcc = -10000
                     maxTestAccEpoch = i
+                elif (self.bonsaiObj.isRegression == True):
                     maxTestAcc = testAcc
-                    self.saveParams(currDir)
+                    maxTestAccEpoch = i
+
+            else:
+                if (self.bonsaiObj.isRegression == False):
+                    if maxTestAcc <= testAcc:
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+                elif (self.bonsaiObj.isRegression == True):
+                    print("Minimum Training MAE : ", np.mean(maxTestAcc))
+                    if maxTestAcc >= testAcc:
+                        # For regression , we're more interested in the minimum MAE.
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+            if(i == totalEpochs-1):
+                if (self.bonsaiObj.isRegression == True):
+                    print("TESTING")
+                    print("-------------------------------------------")
+                    pred = pred[0]
+                    print(np.transpose(pred).shape, Ytest.shape)
+                    print("R2 : ", r2_score(Ytest, np.transpose(pred)))
+                    print("Test MAE : ", mean_absolute_error(Ytest, np.transpose(pred)))
```

Comment:
There is something wrong with one of the two MAE calculations. PFA the two different values for the last epoch on usps10 classification dataset posed as regression.
![capture](https://user-images.githubusercontent.com/17122157/47268416-5856c300-d56e-11e8-84c4-359b76054db8.JPG)

Reply:
This will be resolved automatically once I remove the 'if' condition pertaining to the last epoch.